calyx-ir/src/builder.rs

```diff
@@ -162,7 +162,12 @@ impl<'a> Builder<'a> {
     pub fn add_constant(&mut self, val: u64, width: u64) -> RRC<ir::Cell> {
         // Ensure that the value can fit within the width
         assert!(
-            val < (1 << width),
+            val < match width.cmp(&64) {
+                cmp::Ordering::Less => 1 << width,
+                cmp::Ordering::Equal => u64::MAX,
```

Comment:
This could probably be a `<=` in the `u64` case, but I'm not sure that matters much! Doesn't seem like people want to use `u64::MAX` (all-1s) all that often…